### PR TITLE
:warning: make UseExistingCluster a pointer to support explicitly op…

### PR DESCRIPTION
**Pull request by adohe**
…t-out in the code

just as what #461 mentioned, make UseExistingCluster a pointer thus gives a way to explicitly opt-out in the code.

### CODE REVIEW

File: pkg/envtest/server.go [hunk 1]
```diff
@@ -256,3 +253,11 @@ func (te *Environment) defaultTimeouts() error {
 	}
 	return nil
 }
+
+func (te *Environment) useExistingCluster() bool {
+	if te.UseExistingCluster == nil {
+		useExistingCluster := strings.ToLower(os.Getenv(envUseExistingCluster)) == "true"
+		te.UseExistingCluster = &useExistingCluster
```
Comment (DirectXMan12) [inline pkg/envtest/server.go]: no need to populate this here -- just return the env var without populating

Comment (adohe) [reply pkg/envtest/server.go]: comment addressed, ptal

File: pkg/envtest/server.go [hunk 2]
```diff
@@ -256,3 +252,10 @@ func (te *Environment) defaultTimeouts() error {
 	}
 	return nil
 }
+
+func (te *Environment) useExistingCluster() bool {
+	if te.UseExistingCluster == nil {
+		return strings.ToLower(os.Getenv(envUseExistingCluster)) == "true"
+	}
+	return *te.UseExistingCluster == true
```
Comment (DirectXMan12) [inline pkg/envtest/server.go]: nit: no need for `== true` here, since it's a bool

**Comment by adohe** [comment]
comment addressed and CI happy @DirectXMan12 ptal

**Pull request merged**